ImageProcessing/README.md

```diff
@@ -0,0 +1,747 @@
+
+1. At this point, your `rider-photo-state-machine.json` file should look like this (the AWS Lambda ARNs are examples): 
+	<details>
+	<summary><strong>(expand to see)</strong></summary><p>
+
+	```JSON
+	{
+	  "Comment": "Rider photo processing workflow",
+	  "StartAt": "FaceDetection",
+	  "States": {
+	    "FaceDetection": {
+	      "Type": "Task",
+	      "Resource": "arn:aws:lambda:us-west-2:012345678912:function:wild-ryde-step-module-FaceDetectionFunction-4AYSKX2EGPV0",
+	      "ResultPath": "$.detectedFaceDetails",
+	      "Next": "CheckFaceDuplicate",
+	      "Catch": [
+	        {
+	          "ErrorEquals": [
+	            "PhotoDoesNotMeetRequirementError"
+	          ],
+	          "ResultPath": "$.errorInfo",
+	          "Next": "PhotoDoesNotMeetRequirement"
+	        }
+	      ]
+	    },
+	    "PhotoDoesNotMeetRequirement": {
+	      "Type": "Task",
+	      "Resource": "arn:aws:lambda:us-west-2:012345678912:function:wild-ryde-step-module-NotificationPlaceholderFunct-CDRLZC8BRFWP",
+	      "End": true
+	    },
+	    "CheckFaceDuplicate": {
+	      "Type": "Task",
+	      "Resource": "arn:aws:lambda:us-west-2:012345678912:function:wild-ryde-step-module-FaceSearchFunction-1IT67V4J214DC",
+	      "ResultPath": null,
+	      "Next": "ParallelProcessing",
+	      "Catch": [
+	        {
+	          "ErrorEquals": [
+	            "FaceAlreadyExistsError"
+	          ],
+	          "ResultPath": "$.errorInfo",
+	          "Next": "PhotoDoesNotMeetRequirement"
+	        }
+	      ]
+	    },
+	    "ParallelProcessing": {
+	      "Type": "Parallel",
+	      "Branches": [
+	        {
+	          "StartAt": "AddFaceToIndex",
+	          "States": {
+	            "AddFaceToIndex": {
+	              "Type": "Task",
+	              "Resource": "arn:aws:lambda:us-west-2:012345678912:function:wild-ryde-step-module-IndexFaceFunction-15658V8WUI67V",
+	              "End": true
+	            }
+	          }
+	        },
+	        {
+	          "StartAt": "Thumbnail",
+	          "States": {
+	            "Thumbnail": {
+	              "Type": "Task",
+	              "Resource": "arn:aws:lambda:us-west-2:012345678912:function:wild-ryde-step-module-ThumbnailFunction-A30TCJMIG0U8",
+	              "End": true
+	            }
+	          }
+	        }
+	      ],
+	      "ResultPath": "$.parallelResult",
+	      "Next": "PersistMetadata"
+	    },
+	    "PersistMetadata": {
+	      "Type": "Task",
+	      "Resource": "arn:aws:lambda:us-west-2:012345678912:function:wild-ryde-step-module-PersistMetadataFunction-9PDCT2DT7K70",
+	      "ResultPath": null,
+	      "End": true
+	    }
+	  }
+	}	
+	```
+	</p></details>
+
+1. Go back the AWS Step Functions Console, create a new state machine `RiderPhotoProcessing-4` by copy-pasting the updated JSON definition:
+
+	![create state machine with persistence step](./images/create-machine-with-persistence.png)
+	
+1. Test the new state machine with test input:
```

Comment:
After adding the `ResultPath` in the longer snippet, it stills fails but with a different error:

```
2017-05-17T02:00:52.375Z a69a0acf-3aa4-11e7-b535-abe52cfd2038 TypeError: Cannot read property 'FaceId' of undefined at exports.handler.docClient.put.promise.then.data.catch (/var/task/index.js:22:40)
```